src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParseException.cs

```diff
@@ -6,22 +6,32 @@
 namespace System.Text.RegularExpressions
 {
     [Serializable]
-    internal sealed class RegexParseException : ArgumentException
+    public sealed class RegexParseException : ArgumentException
     {
-        private readonly RegexParseError _error; // tests access this via private reflection
-
         /// <summary>Gets the error that happened during parsing.</summary>
-        public RegexParseError Error => _error;
+        public RegexParseError Error { get; }
 
         /// <summary>Gets the offset in the supplied pattern.</summary>
         public int Offset { get; }
 
-        public RegexParseException(RegexParseError error, int offset, string message) : base(message)
+        public RegexParseException(RegexParseError error, int offset)
+        {
+            Error = error;
+            Offset = offset;
+        }
+
+        internal RegexParseException(RegexParseError error, int offset, string message) : base(message)
         {
-            _error = error;
+            Error = error;
             Offset = offset;
```

Comment:
Check offset >= 0 or throw?

Reply:
And need a test for that

Reply:
And verify Error is valid of course (in C# enums can have arbitrary values so I think we generally verify them when passed in. Even here where nobody is likely  going to create the exception themselves)

Reply:
In the meeting, they suggested that if users create the exception, they should be allowed to put any rubbish in there. They were against throwing on the ctor of the exception, basically.

Reply:
I've changed it to `> 0`, as it is never `0`.